compiler/hash-typecheck/src/ops/pats.rs

```diff
@@ -345,6 +337,7 @@ impl<'tc> PatMatcher<'tc> {
             let new_pat_members =
                 self.builder().create_pat_args(filtered_members, ParamOrigin::Tuple);
 
+            // @@Todo: Remove this modification !
```

Comment:
Why?

Reply:
Well, it would be good to avoid arbitrarily modifying terms and patterns; a lot of places expect that terms and patterns are immutable. It could also mess with error reporting in some cases.

Reply:
It is better to create new terms/patterns each time instead of modifying, at least for now..

Reply:
If this modification is avoided, all these operations now need to account for possible overwrites for these patterns. Without this modification, the code will be pretty messy since it would have to "bubble up" all the way to where this is used within the traversal logic.

Besides, the patterns that are being modified are useless without this modification and no one needs to keep the original variant.

Reply:
Will create a separate PR for this